spec/unit/resource_spec.rb

```diff
@@ -906,6 +906,26 @@ def inject_and_set_defaults(resource, scope)
     # Note: to_stringified_spec.rb has tests for all other data types
   end
 
+  describe 'when serializing resources' do
+    require 'puppet_spec/compiler'
+    include PuppetSpec::Compiler
+
+    it 'should do something' do
```

Comment:
Could you break this up into two tests?

```ruby
    it 'serializes rich data' do
      resource = compile_to_catalog('notify {"foo": message => Deferred("func", ["a", "b", "c"])}')
      expect(resource.to_data_hash).to eq({ ... })
    end
```

Reply:
Yes, sorry. That's a crappy test title as well. I missed that when I was cleaning up the code from my experiments.